src/Chapter3.hs

```diff
@@ -374,6 +380,26 @@ after the fight. The battle has the following possible outcomes:
 
 -}
 
+data Monster = Monster
+  { monsterHealth :: Int,
+    monsterAttack :: Int,
+    monsterGold   :: Int
+  }
+  deriving (Show)
+
+data Knight = Knight
+  { knightHealth :: Int,
+    knightAttack :: Int,
+    knightGold   :: Int
+  }
+  deriving (Show)
+
+fight :: Monster -> Knight -> Int
+fight m k
+  | knightAttack k >= monsterAttack m = knightGold k + monsterGold m
```

Comment:
If the knight attack is bigger than monsters, it still doesn't mean, than the monster would be defeated. In fact, it is better to compare to health to see if it is enough to kill the monster ⚔️